community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1792,7 +1811,49 @@ def nodeset_reservation(self, nodeset: object) -> Optional[ReservationDetails]:
             zone=zone,
             name=name,
             policies=policies,
-            bulk_insert_name=nodeset.reservation_name)
+            bulk_insert_name=reservation_name)
+    
+    def future_reservation(self, futureReservation:str, zone_policy_allow: Optional[list]) -> Optional[FutureReservation]:
+        zones = list(zone_policy_allow or [])
+        assert len(zones) == 1, "Only single zone is supported if using a reservation"
+        zone = zones[0]
+        
+        project = self.project
+        name = futureReservation
+        nodesAreActive = False
+        specificReservationRequired = None
+        activeReservation = None
+        endTime = None
+        startTime = None
+        
+        regex = re.compile(r'/futureReservations/(?P<name>[^/]+)(/.*)?$')
+
+        if match := regex.search(futureReservation):
+            name = match.group("name")
+
+        if reservation := self._get_future_reservation(project, zone, name):
+            startString = reservation.get("timeWindow").get("startTime")
+            endString = reservation.get("timeWindow").get("endTime")
+            startTime = datetime.fromisoformat(startString[:len(startString)-1])
+            endTime = datetime.fromisoformat(endString[:len(endString)-1])
```

Comment:
* Why trim, what at the end?
* There is a better way to trim last character
```py
>>> "hello"[:-1]
'hell'
```

Reply:
The trim is because the timestamp ends with a Z (to indicate UTC) but datetime doesn't handle the Z portion (I think in their python 3.12 package they add this feature) so I just trim it.